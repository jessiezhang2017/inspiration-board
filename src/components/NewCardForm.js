import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: '',
      emoji: '',
    };
  }

  resetState = () => {
    this.setState({
      text: '',
      emoji: '',

    });
  }

  onFormChange = (event) => {
    const field = event.target.name;
    const value = event.target.value;

    const updatedState = {};
    updatedState[field] = value;
    this.setState(updatedState);
  }

  onSubmit = (event) => {
    console.log(event);
    event.preventDefault();
    const {text, emoji} = this.state;

    if (text === '') return;

    console.log(event);
    this.props.addCardCallback(this.state);
    this.resetState();
  }

  render() {
    return (
      <form onSubmit={this.onSubmit} name="new-card-form" id="new-card-form" className="new-card-form">
        <div>
          <label className="new-card-form__form-label" htmlFor="text">Text</label>
          <input name="text" onChange={this.onFormChange} value={this.state.text} />
        </div>
        <div>
          <label className="new-card-form__form-label " htmlFor="emoji">Emoji</label>
          <input name="emoji"  onChange={this.onFormChange} value={this.state.emoji} />
        </div>

        <input className="new-card-form__form-button " type="submit" name="submit" value="Add a Card" />
      </form>
    );
  }

}

NewCardForm.propTypes = {
  addCardCallback: PropTypes.func.isRequired,
};

export default NewCardForm;
