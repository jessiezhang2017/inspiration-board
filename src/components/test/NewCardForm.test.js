import React from 'react';
import NewCardForm from '../NewCardForm';
import {shallow} from 'enzyme';


describe('newcardform',()=>{
  it('will match the newcardform Snapshot', ()=>{
    const wrapper = shallow(<NewCardForm
      addCardCallback={()=>{}}

      />);
    expect(wrapper).toMatchSnapshot();
  });


});
