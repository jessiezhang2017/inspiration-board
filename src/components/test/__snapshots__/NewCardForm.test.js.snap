// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newcardform will match the newcardform Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    addCardCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <label
            className="new-card-form__form-label"
            htmlFor="text"
          >
            Text
          </label>
          <input
            name="text"
            onChange={[Function]}
            value=""
          />
        </div>,
        <div>
          <label
            className="new-card-form__form-label "
            htmlFor="emoji"
          >
            Emoji
          </label>
          <input
            name="emoji"
            onChange={[Function]}
            value=""
          />
        </div>,
        <input
          className="new-card-form__form-button "
          name="submit"
          type="submit"
          value="Add a Card"
        />,
      ],
      "className": "new-card-form",
      "id": "new-card-form",
      "name": "new-card-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="new-card-form__form-label"
              htmlFor="text"
            >
              Text
            </label>,
            <input
              name="text"
              onChange={[Function]}
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Text",
              "className": "new-card-form__form-label",
              "htmlFor": "text",
            },
            "ref": null,
            "rendered": "Text",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "text",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="new-card-form__form-label "
              htmlFor="emoji"
            >
              Emoji
            </label>,
            <input
              name="emoji"
              onChange={[Function]}
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Emoji",
              "className": "new-card-form__form-label ",
              "htmlFor": "emoji",
            },
            "ref": null,
            "rendered": "Emoji",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "emoji",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "new-card-form__form-button ",
          "name": "submit",
          "type": "submit",
          "value": "Add a Card",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <label
              className="new-card-form__form-label"
              htmlFor="text"
            >
              Text
            </label>
            <input
              name="text"
              onChange={[Function]}
              value=""
            />
          </div>,
          <div>
            <label
              className="new-card-form__form-label "
              htmlFor="emoji"
            >
              Emoji
            </label>
            <input
              name="emoji"
              onChange={[Function]}
              value=""
            />
          </div>,
          <input
            className="new-card-form__form-button "
            name="submit"
            type="submit"
            value="Add a Card"
          />,
        ],
        "className": "new-card-form",
        "id": "new-card-form",
        "name": "new-card-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="new-card-form__form-label"
                htmlFor="text"
              >
                Text
              </label>,
              <input
                name="text"
                onChange={[Function]}
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Text",
                "className": "new-card-form__form-label",
                "htmlFor": "text",
              },
              "ref": null,
              "rendered": "Text",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "text",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="new-card-form__form-label "
                htmlFor="emoji"
              >
                Emoji
              </label>,
              <input
                name="emoji"
                onChange={[Function]}
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Emoji",
                "className": "new-card-form__form-label ",
                "htmlFor": "emoji",
              },
              "ref": null,
              "rendered": "Emoji",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "emoji",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-card-form__form-button ",
            "name": "submit",
            "type": "submit",
            "value": "Add a Card",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
